{"expand":"renderedFields,names,schema,transitions,operations,editmeta,changelog","id":"12399953","self":"https://issues.apache.org/jira/rest/api/latest/issue/12399953","key":"DERBY-3773","fields":{"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/1","id":"1","description":"A problem which impairs or prevents the functions of the product.","iconUrl":"https://issues.apache.org/jira/images/icons/issuetypes/bug.png","name":"Bug","subtask":false},"timespent":null,"project":{"self":"https://issues.apache.org/jira/rest/api/2/project/10594","id":"10594","key":"DERBY","name":"Derby","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/projectavatar?pid=10594&avatarId=10122","24x24":"https://issues.apache.org/jira/secure/projectavatar?size=small&pid=10594&avatarId=10122","16x16":"https://issues.apache.org/jira/secure/projectavatar?size=xsmall&pid=10594&avatarId=10122","32x32":"https://issues.apache.org/jira/secure/projectavatar?size=medium&pid=10594&avatarId=10122"},"projectCategory":{"self":"https://issues.apache.org/jira/rest/api/2/projectCategory/10090","id":"10090","description":"DB related projects","name":"DB"}},"fixVersions":[{"self":"https://issues.apache.org/jira/rest/api/2/version/12313345","id":"12313345","description":"","name":"10.4.2.0","archived":false,"released":true,"releaseDate":"2008-09-05"},{"self":"https://issues.apache.org/jira/rest/api/2/version/12313771","id":"12313771","description":"build: 764942, 14/Apr/09. Voted release: 28/Apr/09. Announced: 1/May/09","name":"10.5.1.1","archived":false,"released":true,"releaseDate":"2009-04-28"}],"aggregatetimespent":null,"resolution":{"self":"https://issues.apache.org/jira/rest/api/2/resolution/1","id":"1","description":"A fix for this issue is checked into the tree and tested.","name":"Fixed"},"customfield_12310220":null,"customfield_12312322":null,"customfield_12312323":null,"customfield_12310420":"23834","customfield_12310222":"1_*:*_1_*:*_157740952_*|*_6_*:*_1_*:*_0","customfield_12312320":null,"customfield_12312321":null,"customfield_12312120":null,"customfield_12312121":null,"resolutiondate":"2008-07-11T16:00:51.504+0000","workratio":-1,"customfield_12312328":null,"customfield_12312329":null,"customfield_12312326":null,"customfield_12310300":null,"customfield_12312327":null,"customfield_12312324":null,"customfield_12312325":null,"lastViewed":null,"watches":{"self":"https://issues.apache.org/jira/rest/api/2/issue/DERBY-3773/watchers","watchCount":0,"isWatching":false},"created":"2008-07-09T20:11:50.552+0000","priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/3","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/major.png","name":"Major","id":"3"},"labels":[],"customfield_12312333":null,"customfield_12312334":null,"customfield_12310310":"0.0","customfield_12312331":null,"customfield_12312332":null,"timeestimate":null,"aggregatetimeoriginalestimate":null,"versions":[{"self":"https://issues.apache.org/jira/rest/api/2/version/12313143","id":"12313143","description":"Head of 10.3 branch","name":"10.3.3.1","archived":false,"released":false},{"self":"https://issues.apache.org/jira/rest/api/2/version/12313345","id":"12313345","description":"","name":"10.4.2.0","archived":false,"released":true,"releaseDate":"2008-09-05"},{"self":"https://issues.apache.org/jira/rest/api/2/version/12313771","id":"12313771","description":"build: 764942, 14/Apr/09. Voted release: 28/Apr/09. Announced: 1/May/09","name":"10.5.1.1","archived":false,"released":true,"releaseDate":"2009-04-28"}],"customfield_12311120":null,"customfield_12312330":null,"issuelinks":[],"customfield_12312339":null,"assignee":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kmarsden","name":"kmarsden","emailAddress":"kmarsdenderby at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kathey Marsden","active":true},"customfield_12312337":null,"customfield_12312338":null,"updated":"2009-05-04T18:22:45.547+0000","customfield_12312335":null,"customfield_12311720":null,"customfield_12312336":null,"status":{"self":"https://issues.apache.org/jira/rest/api/2/status/6","description":"The issue is considered finished, the resolution is correct. Issues which are closed can be reopened.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/closed.png","name":"Closed","id":"6","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Complete"}},"components":[{"self":"https://issues.apache.org/jira/rest/api/2/component/11413","id":"11413","name":"Test"},{"self":"https://issues.apache.org/jira/rest/api/2/component/11414","id":"11414","name":"Tools"}],"timeoriginalestimate":null,"description":"testImportDataExportQueryLobsInExtFile(org.apache.derbyTesting.f\nunctionTests.tests.tools.ImportExportLobTest)junit.framework.Ass\nertionFailedError: Clobs differ at index 1 expected:<99> but \nwas:<196>\n\tat \norg.apache.derbyTesting.junit.BaseJDBCTestCase.assertEquals(Base\nJDBCTestCase.java:693)\n\tat \norg.apache.derbyTesting.junit.JDBC.assertSameContents(JDBC.java:\n1139)\n\tat \norg.apache.derbyTesting.functionTests.tests.tools.ImportExportLo\nbTest.verifyData(ImportExportLobTest.java:534)\n\tat \norg.apache.derbyTesting.functionTests.tests.tools.ImportExportLo\nbTest.testImportDataExportQueryLobsInExtFile(ImportExportLobTest\n.java:341)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native \nMethod)\n\tat \nsun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessor\nImpl.java:59)\n\tat \nsun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethod\nAccessorImpl.java:39)\n\tat \norg.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.\njava:101)\n\tat \njunit.extensions.TestDecorator.basicRun(TestDecorator.java:22)\n\tat junit.extensions.TestSetup$1.protect(TestSetup.java:19)\n\tat junit.extensions.TestSetup.run(TestSetup.java:23)\n\tat \njunit.extensions.TestDecorator.basicRun(TestDecorator.java:22)\n\tat junit.extensions.TestSetup$1.protect(TestSetup.java:19)\n\tat junit.extensions.TestSetup.run(TestSetup.java:23)\n\nI ran on Windows with -Dfile.encoding=\"UTF-16\" and got a different error.\n1) testImportDataExportQueryLobsInExtFile(org.apache.derbyTesting.functionTests.tests.tools.ImportExportLobTest)java.sql\n.SQLException: The exception 'java.sql.SQLException: Column 'COLUMN4' is either not in any table in the FROM list or app\nears within a join specification and is outside the scope of the join specification or appears in a HAVING clause and is\n not in the GROUP BY list. If this is a CREATE or ALTER TABLE  statement then 'COLUMN4' is not a column in the target ta\nble.' was thrown while evaluating an expression.\n        at org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:95)\n        at org.apache.derby.impl.jdbc.Util.newEmbedSQLException(Util.java:87)\n        at org.apache.derby.impl.jdbc.Util.seeNextException(Util.java:223)\n        at org.apache.derby.impl.jdbc.TransactionResourceImpl.wrapInSQLException(TransactionResourceImpl.java:398)\n        at org.apache.derby.impl.jdbc.TransactionResourceImpl.handleException(TransactionResourceImpl.java:346)\n        at org.apache.derby.impl.jdbc.EmbedConnection.handleException(EmbedConnection.java:2183)\n        at org.apache.derby.impl.jdbc.ConnectionChild.handleException(ConnectionChild.java:81)\n        at org.apache.derby.impl.jdbc.EmbedStatement.executeStatement(EmbedStatement.java:1325)\n        at org.apache.derby.impl.jdbc.EmbedPreparedStatement.executeStatement(EmbedPreparedStatement.java:1652)\n        at org.apache.derby.impl.jdbc.EmbedPreparedStatement.execute(EmbedPreparedStatement.java:1307)\n        at org.apache.derbyTesting.functionTests.tests.tools.ImportExportBaseTest.doImportDataLobsFromExtFile(ImportExpo\nrtBaseTest.java:258)\n        at org.apache.derbyTesting.functionTests.tests.tools.ImportExportLobTest.testImportDataExportQueryLobsInExtFile(\nImportExportLobTest.java:339)\n        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)\n        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)\n        at org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:104)\n        at junit.extensions.TestDecorator.basicRun(TestDecorator.java:22)\n        at junit.extensions.TestSetup$1.protect(TestSetup.java:19)\n        at junit.extensions.TestSetup.run(TestSetup.java:23)\n        at junit.extensions.TestDecorator.basicRun(TestDecorator.java:22)\n        at junit.extensions.TestSetup$1.protect(TestSetup.java:19)\n        at junit.extensions.TestSetup.run(TestSetup.java:23)\n        at org.apache.derbyTesting.junit.BaseTestSetup.run(BaseTestSetup.java:57)\nCaused by: java.sql.SQLException: The exception 'java.sql.SQLException: Column 'COLUMN4' is either not in any table in t\nhe FROM list or appears within a join specification and is outside the scope of the join specification or appears in a H\nAVING clause and is not in the GROUP BY list. If this is a CREATE or ALTER TABLE  statement then 'COLUMN4' is not a colu\nmn in the target table.' was thrown while evaluating an expression.\n        at org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(SQLExceptionFactory.java:45)\n        at org.apache.derby.impl.jdbc.SQLExceptionFactory40.wrapArgsForTransportAcrossDRDA(SQLExceptionFactory40.java:11\n9)\n        at org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:70)\n        ... 36 more\nCaused by: java.sql.SQLException: Column 'COLUMN4' is either not in any table in the FROM list or appears within a join\nspecification and is outside the scope of the join specification or appears in a HAVING clause and is not in the GROUP B\nY list. If this is a CREATE or ALTER TABLE  statement then 'COLUMN4' is not a column in the target table.\n        at org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(SQLExceptionFactory.java:45)\n        at org.apache.derby.impl.jdbc.SQLExceptionFactory40.wrapArgsForTransportAcrossDRDA(SQLExceptionFactory40.java:11\n9)\n        at org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:70)\n        at org.apache.derby.impl.jdbc.Util.generateCsSQLException(Util.java:201)\n        at org.apache.derby.impl.jdbc.TransactionResourceImpl.wrapInSQLException(TransactionResourceImpl.java:391)\n        at org.apache.derby.impl.jdbc.TransactionResourceImpl.handleException(TransactionResourceImpl.java:346)\n        at org.apache.derby.impl.jdbc.EmbedConnection.handleException(EmbedConnection.java:2183)\n        at org.apache.derby.impl.jdbc.ConnectionChild.handleException(ConnectionChild.java:81)\n        at org.apache.derby.impl.jdbc.EmbedPreparedStatement.<init>(EmbedPreparedStatement.java:146)\n        at org.apache.derby.impl.jdbc.EmbedPreparedStatement20.<init>(EmbedPreparedStatement20.java:82)\n        at org.apache.derby.impl.jdbc.EmbedPreparedStatement30.<init>(EmbedPreparedStatement30.java:63)\n        at org.apache.derby.impl.jdbc.EmbedPreparedStatement40.<init>(EmbedPreparedStatement40.java:40)\n        at org.apache.derby.jdbc.Driver40.newEmbedPreparedStatement(Driver40.java:105)\n        at org.apache.derby.impl.jdbc.EmbedConnection.prepareStatement(EmbedConnection.java:1533)\n        at org.apache.derby.impl.jdbc.EmbedConnection.prepareStatement(EmbedConnection.java:1361)\n        at org.apache.derby.impl.load.Import.performImport(Import.java:274)\n        at org.apache.derby.impl.load.Import.importData(Import.java:166)\n        at org.apache.derby.catalog.SystemProcedures.SYSCS_IMPORT_DATA_LOBS_FROM_EXTFILE(SystemProcedures.java:1538)\n        at org.apache.derby.exe.acaa7ac093x011bx0970x9cd2x00003dcd3cb42.g0(Unknown Source)\n        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)\n        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)\n        at org.apache.derby.impl.services.reflect.ReflectMethod.invoke(ReflectMethod.java:46)\n        at org.apache.derby.impl.sql.execute.CallStatementResultSet.open(CallStatementResultSet.java:76)\n        at org.apache.derby.impl.sql.GenericPreparedStatement.execute(GenericPreparedStatement.java:384)\n        at org.apache.derby.impl.jdbc.EmbedStatement.executeStatement(EmbedStatement.java:1235)\n        ... 29 more\nCaused by: ERROR 42X04: Column 'COLUMN4' is either not in any table in the FROM list or appears within a join specificat\nion and is outside the scope of the join specification or appears in a HAVING clause and is not in the GROUP BY list. If\n this is a CREATE or ALTER TABLE  statement then 'COLUMN4' is not a column in the target table.\n        at org.apache.derby.iapi.error.StandardException.newException(StandardException.java:286)\n        at org.apache.derby.impl.sql.compile.ColumnReference.bindExpression(ColumnReference.java:354)\n        at org.apache.derby.impl.sql.compile.ResultColumn.bindExpression(ResultColumn.java:588)\n        at org.apache.derby.impl.sql.compile.ResultColumnList.bindExpressions(ResultColumnList.java:693)\n        at org.apache.derby.impl.sql.compile.SelectNode.bindExpressions(SelectNode.java:443)\n        at org.apache.derby.impl.sql.compile.SelectNode.bindExpressionsWithTables(SelectNode.java:578)\n        at org.apache.derby.impl.sql.compile.DMLStatementNode.bindExpressionsWithTables(DMLStatementNode.java:251)\n        at org.apache.derby.impl.sql.compile.DMLStatementNode.bindResultSetsWithTables(DMLStatementNode.java:175)\n        at org.apache.derby.impl.sql.compile.InsertNode.bindStatement(InsertNode.java:242)\n        at org.apache.derby.impl.sql.GenericStatement.prepMinion(GenericStatement.java:314)\n        at org.apache.derby.impl.sql.GenericStatement.prepare(GenericStatement.java:88)\n        at org.apache.derby.impl.sql.conn.GenericLanguageConnectionContext.prepareInternalStatement(GenericLanguageConne\nctionContext.java:794)\n        at org.apache.derby.impl.jdbc.EmbedPreparedStatement.<init>(EmbedPreparedStatement.java:128)\n        ... 47 more\n2) testImportDataExportQueryLobsInExtFile(org.apache.derbyTesting.functionTests.tests.tools.ImportExportLobTest)java.sql\n.SQLException: The exception 'java.sql.SQLException: Column 'COLUMN4' is either not in any table in the FROM list or app\nears within a join specification and is outside the scope of the join specification or appears in a HAVING clause and is\n not in the GROUP BY list. If this is a CREATE or ALTER TABLE  statement then 'COLUMN4' is not a column in the target ta\nble.' was thrown while evaluating an expression.\n        at org.apache.derby.client.am.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:96)\n        at org.apache.derby.client.am.SqlException.getSQLException(SqlException.java:362)\n        at org.apache.derby.client.am.PreparedStatement.execute(PreparedStatement.java:1588)\n        at org.apache.derbyTesting.functionTests.tests.tools.ImportExportBaseTest.doImportDataLobsFromExtFile(ImportExpo\nrtBaseTest.java:258)\n        at org.apache.derbyTesting.functionTests.tests.tools.ImportExportLobTest.testImportDataExportQueryLobsInExtFile(\nImportExportLobTest.java:339)\n        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)\n        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)\n        at org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:104)\n        at junit.extensions.TestDecorator.basicRun(TestDecorator.java:22)\n        at junit.extensions.TestSetup$1.protect(TestSetup.java:19)\n        at junit.extensions.TestSetup.run(TestSetup.java:23)\n        at org.apache.derbyTesting.junit.BaseTestSetup.run(BaseTestSetup.java:57)\n        at junit.extensions.TestDecorator.basicRun(TestDecorator.java:22)\n        at junit.extensions.TestSetup$1.protect(TestSetup.java:19)\n        at junit.extensions.TestSetup.run(TestSetup.java:23)\n        at junit.extensions.TestDecorator.basicRun(TestDecorator.java:22)\n        at junit.extensions.TestSetup$1.protect(TestSetup.java:19)\n        at junit.extensions.TestSetup.run(TestSetup.java:23)\n        at junit.extensions.TestDecorator.basicRun(TestDecorator.java:22)\n        at junit.extensions.TestSetup$1.protect(TestSetup.java:19)\n        at junit.extensions.TestSetup.run(TestSetup.java:23)\n        at org.apache.derbyTesting.junit.BaseTestSetup.run(BaseTestSetup.java:57)\nCaused by: org.apache.derby.client.am.SqlException: The exception 'java.sql.SQLException: Column 'COLUMN4' is either not\n in any table in the FROM list or appears within a join specification and is outside the scope of the join specification\n or appears in a HAVING clause and is not in the GROUP BY list. If this is a CREATE or ALTER TABLE  statement then 'COLU\nMN4' is not a column in the target table.' was thrown while evaluating an expression.\n        at org.apache.derby.client.am.Statement.completeExecute(Statement.java:1601)\n        at org.apache.derby.client.net.NetStatementReply.parseEXCSQLSTTreply(NetStatementReply.java:322)\n        at org.apache.derby.client.net.NetStatementReply.readExecuteCall(NetStatementReply.java:106)\n        at org.apache.derby.client.net.StatementReply.readExecuteCall(StatementReply.java:75)\n        at org.apache.derby.client.net.NetStatement.readExecuteCall_(NetStatement.java:179)\n        at org.apache.derby.client.am.Statement.readExecuteCall(Statement.java:1567)\n        at org.apache.derby.client.am.PreparedStatement.flowExecute(PreparedStatement.java:2151)\n        at org.apache.derby.client.am.PreparedStatement.executeX(PreparedStatement.java:1594)\n        at org.apache.derby.client.am.PreparedStatement.execute(PreparedStatement.java:1579)\n        ... 38 more\nCaused by: org.apache.derby.client.am.SqlException: Column 'COLUMN4' is either not in any table in the FROM list or appe\nars within a join specification and is outside the scope of the join specification or appears in a HAVING clause and is\nnot in the GROUP BY list. If this is a CREATE or ALTER TABLE  statement then 'COLUMN4' is not a column in the target tab\nle.\n        ... 47 more","customfield_10010":null,"timetracking":{},"customfield_12312026":null,"customfield_12310200":null,"customfield_12312023":null,"customfield_12312024":null,"customfield_12312340":null,"aggregatetimeestimate":null,"customfield_12312022":null,"customfield_12312341":null,"customfield_12312220":null,"customfield_12310921":null,"customfield_12310920":"39228","summary":"ImportExportLobTest failed on Zos  Clobs differ at index 1 expected:<99> but was:<196>","creator":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kmarsden","name":"kmarsden","emailAddress":"kmarsdenderby at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kathey Marsden","active":true},"subtasks":[],"customfield_12310090":null,"reporter":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kmarsden","name":"kmarsden","emailAddress":"kmarsdenderby at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kathey Marsden","active":true},"customfield_12310291":null,"customfield_12310290":null,"aggregateprogress":{"progress":0,"total":0},"environment":"IBM 1.5\n","customfield_12311020":null,"customfield_12310050":null,"duedate":null,"customfield_12310250":null,"progress":{"progress":0,"total":0},"comment":{"startAt":0,"maxResults":0,"total":0,"comments":[]},"votes":{"self":"https://issues.apache.org/jira/rest/api/2/issue/DERBY-3773/votes","votes":0,"hasVoted":false},"worklog":{"startAt":0,"maxResults":20,"total":0,"worklogs":[]},"customfield_12311820":"0|i072un:"}}